src/Nox.Lib/Presentation/Api/OData/ApiRouteEtagOperationFilter.cs

```diff
@@ -0,0 +1,28 @@
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Nox.Presentation.Api.OData;
+
+public class ApiRouteEtagOperationFilter:  IOperationFilter
+{
+    private  static readonly  string[] _eTagRequiredHttpMethods = { HttpMethod.Delete.Method, HttpMethod.Patch.Method, HttpMethod.Put.Method };
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
+    {
+        if (_eTagRequiredHttpMethods.Contains(context.ApiDescription.HttpMethod))
+        {
+            operation.Parameters.Add(new OpenApiParameter
+            {
+                Name = "etag",
```

Comment:
the name should be "If-Match"? this is what is expected on the header?

Reply:
Desired changes have been applied